File: engine/src/main/java/de/hanno/hpengine/engine/graphics/renderer/GLU.java
package de.hanno.hpengine.engine.graphics.renderer;

import org.lwjgl.opengl.GL11;

import static org.lwjgl.opengl.ARBImaging.GL_TABLE_TOO_LARGE;
import static org.lwjgl.opengl.GL11.*;
import static org.lwjgl.opengl.GL30.GL_INVALID_FRAMEBUFFER_OPERATION;

public class GLU {

    public static String gluErrorString(int error_code) {
        switch(error_code) {
            case 100900:
                return "Invalid enum (glu)";
            case 100901:
                return "Invalid value (glu)";
            case 100902:
                return "Out of memory (glu)";
            default:
                return de.hanno.hpengine.util.Util.translateGLErrorString(error_code);
        }
    }

    public static String translateGLErrorString(int error_code) {
        switch (error_code) {
            case GL11.GL_NO_ERROR:
                return "No error";
            case GL_INVALID_ENUM:
                return "Invalid enum";
            case GL_INVALID_VALUE:
                return "Invalid value";
            case GL_INVALID_OPERATION:
                return "Invalid operation";
            case GL_STACK_OVERFLOW:
                return "Stack overflow";
            case GL_STACK_UNDERFLOW:
                return "Stack underflow";
            case GL_OUT_OF_MEMORY:
                return "Out of memory";
            case GL_TABLE_TOO_LARGE:
                return "Table too large";
            case GL_INVALID_FRAMEBUFFER_OPERATION:
                return "Invalid framebuffer operation";
            default:
                return null;
        }
    }
}

File: engine/src/main/kotlin/de/hanno/hpengine/graphics/renderer/GLU.kt
package de.hanno.hpengine.graphics.renderer

import de.hanno.hpengine.util.Util
import org.lwjgl.opengl.ARBImaging
import org.lwjgl.opengl.GL11
import org.lwjgl.opengl.GL30

object GLU {
    fun gluErrorString(error_code: Int): String = when (error_code) {
        100900 -> "Invalid enum (glu)"
        100901 -> "Invalid value (glu)"
        100902 -> "Out of memory (glu)"
        else -> Util.translateGLErrorString(error_code)
    }

    fun translateGLErrorString(error_code: Int): String? = when (error_code) {
        GL11.GL_NO_ERROR -> "No error"
        GL11.GL_INVALID_ENUM -> "Invalid enum"
        GL11.GL_INVALID_VALUE -> "Invalid value"
        GL11.GL_INVALID_OPERATION -> "Invalid operation"
        GL11.GL_STACK_OVERFLOW -> "Stack overflow"
        GL11.GL_STACK_UNDERFLOW -> "Stack underflow"
        GL11.GL_OUT_OF_MEMORY -> "Out of memory"
        ARBImaging.GL_TABLE_TOO_LARGE -> "Table too large"
        GL30.GL_INVALID_FRAMEBUFFER_OPERATION -> "Invalid framebuffer operation"
        else -> null
    }
}